src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1264,28 +1283,139 @@ public static bool IsPatternContext(this SyntaxTree syntaxTree, SyntaxToken left
                 return true;
             }
 
+            // case ($
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken) && node.IsParentKind(SyntaxKind.CaseSwitchLabel))
+            {
+                return true;
+            }
+
             // e switch { $$
             // e switch { ..., $$
-            if (leftToken.IsKind(SyntaxKind.OpenBraceToken, SyntaxKind.CommaToken) && leftToken.Parent.IsKind(SyntaxKind.SwitchExpression))
+            if (leftToken.IsKind(SyntaxKind.OpenBraceToken, SyntaxKind.CommaToken) && node.IsKind(SyntaxKind.SwitchExpression))
             {
                 return true;
             }
 
             // e is ($$
             // e is (..., $$
-            if (leftToken.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken) && leftToken.Parent.IsKind(SyntaxKind.PositionalPatternClause))
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken, SyntaxKind.CommaToken) && node.IsKind(SyntaxKind.PositionalPatternClause))
             {
                 return true;
             }
 
             // e is { P: $$
             // e is { ..., P: $$
-            if (leftToken.IsKind(SyntaxKind.ColonToken) && leftToken.Parent.IsKind(SyntaxKind.NameColon) &&
-                leftToken.Parent.IsParentKind(SyntaxKind.Subpattern))
+            if (leftToken.IsKind(SyntaxKind.ColonToken) && node.IsKind(SyntaxKind.NameColon) &&
+                node.IsParentKind(SyntaxKind.Subpattern))
+            {
+                return true;
+            }
+
+            // e is { P: ($$
+            if (leftToken.IsKind(SyntaxKind.OpenParenToken) && node.IsParentKind(SyntaxKind.ConstantPattern))
+            {
+                return true;
+            }
+
+#if !CODE_STYLE
+            // if (e is { P: (1 or $$
+            if ((leftToken.IsKind(SyntaxKind.AndKeyword) || leftToken.IsKind(SyntaxKind.OrKeyword)) &&
+                node is BinaryPatternSyntax)
             {
                 return true;
             }
```

Comment:
so it would be more ideal to rewrite these as:

```c#
if (leftToken.IsKind(SyntaxKind.AndKeyword) || leftToken.IsKind(SyntaxKind.OrKeyword)
    return node is BinaryPatternSyntax;
```

Think about is this way, once you know you're on an `and/or`, there's no need to perform *other* checks for other syntax kinds.  It's just wasted work.  This allows us to shortcircuit immediately on things. (this applies both above and below this point in the code).